### PR TITLE
rapidyaml 0.7.2 (new formula)

### PR DESCRIPTION
**Pull request by cxwx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/c4core.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@ class C4core < Formula
 
   depends_on "cmake" => [:build, :test]
 
+  conflicts_with "rapidyaml", because: "both install `c4core`"
```
Comment (chenrui333) [inline Formula/c/c4core.rb:21]: the conflicts can be done as followup as syntax only pr

File: Formula/r/rapidyaml.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Rapidyaml < Formula
+  desc "Library to parse and emit YAML, and do it fast"
+  homepage "https://github.com/biojppm/rapidyaml"
+  url "https://github.com/biojppm/rapidyaml/releases/download/v0.7.2/rapidyaml-0.7.2-src.tgz"
+  sha256 "175b71074005a7d48ae03e973f1a7a5c104c588f193ba85044c5b4a97341aae0"
+  license "MIT"
+
+  depends_on "cmake" => :build
+
+  conflicts_with "c4core", because: "both install `c4core`"
```
Comment (chenrui333) [inline Formula/r/rapidyaml.rb:10]: but I did not see the 4core utility in my local though

File: Formula/r/rapidyaml.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", "-DBUILD_SHARED_LIBS=ON", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+  test do
```
Comment (chenrui333) [inline Formula/r/rapidyaml.rb:17]:
```suggestion
  end

  test do
```

File: Formula/r/rapidyaml.rb (hunk 2 above)
Comment (cxwx) [reply Formula/r/rapidyaml.rb:10]:
```
Target /opt/homebrew/include/c4/allocator.hpp
is a symlink belonging to rapidyaml. You can unlink it:
  brew unlink rapidyaml

To force the link and overwrite all conflicting files:
  brew link --overwrite c4core

To list all files that would be deleted:
  brew link --overwrite c4core --dry-run

Possible conflicting files are:
/opt/homebrew/include/c4/allocator.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/allocator.hpp
/opt/homebrew/include/c4/base64.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/base64.hpp
/opt/homebrew/include/c4/bitmask.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/bitmask.hpp
/opt/homebrew/include/c4/blob.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/blob.hpp
/opt/homebrew/include/c4/c4_pop.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/c4_pop.hpp
/opt/homebrew/include/c4/c4_push.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/c4_push.hpp
/opt/homebrew/include/c4/c4core.natvis -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/c4core.natvis
/opt/homebrew/include/c4/char_traits.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/char_traits.hpp
/opt/homebrew/include/c4/charconv.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/charconv.hpp
/opt/homebrew/include/c4/common.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/common.hpp
/opt/homebrew/include/c4/compiler.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/compiler.hpp
/opt/homebrew/include/c4/config.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/config.hpp
/opt/homebrew/include/c4/cpu.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/cpu.hpp
/opt/homebrew/include/c4/ctor_dtor.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ctor_dtor.hpp
/opt/homebrew/include/c4/dump.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/dump.hpp
/opt/homebrew/include/c4/enum.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/enum.hpp
/opt/homebrew/include/c4/error.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/error.hpp
/opt/homebrew/include/c4/export.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/export.hpp
/opt/homebrew/include/c4/ext/debugbreak/debugbreak.h -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ext/debugbreak/debugbreak.h
/opt/homebrew/include/c4/ext/fast_float.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ext/fast_float.hpp
/opt/homebrew/include/c4/ext/fast_float_all.h -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ext/fast_float_all.h
/opt/homebrew/include/c4/ext/rng/rng.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ext/rng/rng.hpp
/opt/homebrew/include/c4/ext/sg14/inplace_function.h -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/ext/sg14/inplace_function.h
/opt/homebrew/include/c4/format.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/format.hpp
/opt/homebrew/include/c4/gcc-4.8.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/gcc-4.8.hpp
/opt/homebrew/include/c4/hash.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/hash.hpp
/opt/homebrew/include/c4/language.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/language.hpp
/opt/homebrew/include/c4/memory_resource.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/memory_resource.hpp
/opt/homebrew/include/c4/memory_util.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/memory_util.hpp
/opt/homebrew/include/c4/platform.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/platform.hpp
/opt/homebrew/include/c4/preprocessor.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/preprocessor.hpp
/opt/homebrew/include/c4/restrict.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/restrict.hpp
/opt/homebrew/include/c4/span.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/span.hpp
/opt/homebrew/include/c4/std/std.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/std.hpp
/opt/homebrew/include/c4/std/std_fwd.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/std_fwd.hpp
/opt/homebrew/include/c4/std/string.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/string.hpp
/opt/homebrew/include/c4/std/string_fwd.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/string_fwd.hpp
/opt/homebrew/include/c4/std/string_view.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/string_view.hpp
/opt/homebrew/include/c4/std/tuple.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/tuple.hpp
/opt/homebrew/include/c4/std/vector.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/vector.hpp
/opt/homebrew/include/c4/std/vector_fwd.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/std/vector_fwd.hpp
/opt/homebrew/include/c4/substr.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/substr.hpp
/opt/homebrew/include/c4/substr_fwd.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/substr_fwd.hpp
/opt/homebrew/include/c4/szconv.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/szconv.hpp
/opt/homebrew/include/c4/type_name.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/type_name.hpp
/opt/homebrew/include/c4/types.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/types.hpp
/opt/homebrew/include/c4/unrestrict.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/unrestrict.hpp
/opt/homebrew/include/c4/utf.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/utf.hpp
/opt/homebrew/include/c4/windows.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/windows.hpp
/opt/homebrew/include/c4/windows_pop.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/windows_pop.hpp
/opt/homebrew/include/c4/windows_push.hpp -> /opt/homebrew/Cellar/rapidyaml/0.7.2/include/c4/windows_push.hpp
Error: Could not symlink lib/cmake/c4core/c4coreConfig.cmake
Target /opt/homebrew/lib/cmake/c4core/c4coreConfig.cmake
```

File: Formula/c/c4core.rb [hunk 4]
```diff
@@ -18,6 +18,8 @@ class C4core < Formula
 
   depends_on "cmake" => [:build, :test]
 
+  conflicts_with "rapidyaml", because: "both install `c4core` files `include/c4`"
```
Comment (chenrui333) [inline Formula/c/c4core.rb:21]: let's do this as followup PR?

**Comment by chenrui333** [comment]
once exclude the conflicts_with message and you can squash all the commits.

**Comment by cxwx** [comment]
> once exclude the conflicts_with message (which can be handled as followup pr) and you can squash all the commits. i think we are good after that.

I've squash the commits, but the check of conflict was failed as it can be handled as followup pr.
It that OK?

File: Formula/r/rapidyaml.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  desc "Library to parse and emit YAML, and do it fast"
+  homepage "https://github.com/biojppm/rapidyaml"
+  url "https://github.com/biojppm/rapidyaml/releases/download/v0.7.2/rapidyaml-0.7.2-src.tgz"
+  sha256 "175b71074005a7d48ae03e973f1a7a5c104c588f193ba85044c5b4a97341aae0"
+  license "MIT"
+
+  depends_on "cmake" => :build
+
+  conflicts_with "c4core", because: "both install `c4core` filse `include/c4`"
+
```
Comment (chenrui333) [inline Formula/r/rapidyaml.rb:11]: this should be also handled as followup pr

**Review by chenrui333 — APPROVED** [review]
after removing the conflicts_with, I think we can shoot it out.

**Review by daeho-ro — APPROVED** [review]

**Comment by chenrui333** [comment]
@cxwx, thanks for your contribution to Homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**